### ISSUE TITLE
Recent changes generate a bunch of errors

### ISSUE THREAD

**Issue by junghans** [votca/ctp#115]
From https://gitlab.com/votca/ctp/pipelines/47122122:
```
[ 35%] Building CXX object ctp/src/libctp/CMakeFiles/votca_ctp.dir/jobcalculatorfactory.cc.o
cd /builds/votca/ctp/votca/build/ctp/src/libctp && /usr/lib64/ccache/g++  -Dvotca_ctp_EXPORTS -I/builds/votca/ctp/votca/build/ctp/include -I/builds/votca/ctp/votca/ctp/include -I/builds/votca/ctp/votca/tools/include -I/builds/votca/ctp/votca/build/tools/include -I/usr/include/eigen3 -I/builds/votca/ctp/votca/csg/include -I/builds/votca/ctp/votca/build/ctp/src/libctp  -Wall -Werror -fopenmp -g -fPIC   -std=c++11 -o CMakeFiles/votca_ctp.dir/jobcalculatorfactory.cc.o -c /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc
In file included from /builds/votca/ctp/votca/ctp/include/votca/ctp/qmthread.h:6,
                 from /builds/votca/ctp/votca/ctp/include/votca/ctp/parallelxjobcalc.h:6,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/idft.h:25,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc:23:
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h: In instantiation of 'void votca::ctp::Ewald<EwaldMethod>::ReadJobFile(votca::ctp::Topology*) [with EwaldMethod = votca::ctp::PEwald3D3D]':
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:118:6:   required from here
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
In file included from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc:27:
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:175:24: error: unused variable 'estat' [-Werror=unused-variable]
                 double estat  = poutput.get("estat").as<double>();
                        ^~~~~
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:176:24: error: unused variable 'eindu' [-Werror=unused-variable]
                 double eindu  = poutput.get("eindu").as<double>();
                        ^~~~~
In file included from /builds/votca/ctp/votca/ctp/include/votca/ctp/qmthread.h:6,
                 from /builds/votca/ctp/votca/ctp/include/votca/ctp/parallelxjobcalc.h:6,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/idft.h:25,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc:23:
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h: In instantiation of 'void votca::ctp::Ewald<EwaldMethod>::ReadJobFile(votca::ctp::Topology*) [with EwaldMethod = votca::ctp::Ewald3D3D]':
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:118:6:   required from here
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
In file included from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc:27:
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:175:24: error: unused variable 'estat' [-Werror=unused-variable]
                 double estat  = poutput.get("estat").as<double>();
                        ^~~~~
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:176:24: error: unused variable 'eindu' [-Werror=unused-variable]
                 double eindu  = poutput.get("eindu").as<double>();
                        ^~~~~
In file included from /builds/votca/ctp/votca/ctp/include/votca/ctp/qmthread.h:6,
                 from /builds/votca/ctp/votca/ctp/include/votca/ctp/parallelxjobcalc.h:6,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/idft.h:25,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc:23:
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h: In instantiation of 'void votca::ctp::Ewald<EwaldMethod>::ReadJobFile(votca::ctp::Topology*) [with EwaldMethod = votca::ctp::Ewald3D2D]':
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:118:6:   required from here
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
In file included from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc:27:
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:175:24: error: unused variable 'estat' [-Werror=unused-variable]
                 double estat  = poutput.get("estat").as<double>();
                        ^~~~~
/builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/ewald.h:176:24: error: unused variable 'eindu' [-Werror=unused-variable]
                 double eindu  = poutput.get("eindu").as<double>();
                        ^~~~~
In file included from /builds/votca/ctp/votca/ctp/include/votca/ctp/qmthread.h:6,
                 from /builds/votca/ctp/votca/ctp/include/votca/ctp/parallelxjobcalc.h:6,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculators/idft.h:25,
                 from /builds/votca/ctp/votca/ctp/src/libctp/jobcalculatorfactory.cc:23:
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: error: the address of '_log' will never be NULL [-Werror=address]
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^
/builds/votca/ctp/votca/ctp/include/votca/ctp/logger.h:35:11: note: in definition of macro 'CTP_LOG'
 if ( &log != NULL && level > (log).getReportLevel() ) ; \
           ^~
cc1plus: all warnings being treated as errors
```

This will prevent us from updating the ctp submodule in votca/votca and hence hide these updates from the users.

**Comment by junghans** [comment]
Fixed in 77392ebc515e0872314ea8a5dcce2a5b05cf71e7

**Comment by junghans** [comment]
Updates are part of https://github.com/votca/votca/pull/126
